lnst/Controller/Controller.py

```diff
@@ -171,7 +171,8 @@ def run(self, recipe, **kwargs):
                                            log_list=self._log_ctl.get_recipe_log_list()))
                 recipe.test()
             except Exception as exc:
-                recipe.current_run.exception = exc
+                if recipe.current_run:
+                    recipe.current_run.exception = exc
```

Comment:
so this can only happen in case `_map_match` or `_init_run` throw an exception, i think the init run case shouldn't be possible, so what kind of exceptions have you seen with `_map_match`?

Reply:
Right, was able to reproduce. So in that case you simply get following message after agent machines are spawned:
```
'NoneType' object has no attribute 'exception'
```

You get no other information than that. With the fix I get full exception log:
```
2023-06-28 17:32:58       (localhost)        -   ERROR: Recipe execution terminated by unexpected exception
2023-06-28 17:32:58       (localhost)        -   DEBUG: 
    Traceback (most recent call last):
      File "/home/igyn/tmp/lnst/lnst/Controller/Controller.py", line 169, in run
        self._map_match(match, req, recipe)
      File "/home/igyn/tmp/lnst/lnst/Controller/Controller.py", line 194, in _map_match
        self._prepare_machine(machine)
      File "/home/igyn/tmp/lnst/lnst/Controller/Controller.py", line 215, in _prepare_machine
        machine.prepare_machine()
      File "/home/igyn/tmp/lnst/lnst/Controller/Machine.py", line 366, in prepare_machine
        self._send_device_classes()
      File "/home/igyn/tmp/lnst/lnst/Controller/Machine.py", line 387, in _send_device_classes
        self.send_class(cls)
      File "/home/igyn/tmp/lnst/lnst/Controller/Machine.py", line 410, in send_class
        self.rpc_call("load_cached_module", module_name, res_hash, netns=netns)
      File "/home/igyn/tmp/lnst/lnst/Controller/Machine.py", line 320, in rpc_call
        return self._msg_dispatcher.send_message(self, msg)
      File "/home/igyn/tmp/lnst/lnst/Controller/MessageDispatcher.py", line 131, in send_message
        self._handle_disconnects(set(connected_agents)-
      File "/home/igyn/tmp/lnst/lnst/Controller/MessageDispatcher.py", line 245, in _handle_disconnects
        raise ConnectionError(msg)
    lnst.Controller.MessageDispatcher.ConnectionError: Agents ['host1'] hard-disconnected from the controller.
```

Reply:
right, makes sense, normally python should now show both exceptions as a sort of:

```
Traceback1
...

during handling of above exception another excpetion occured:

Traceback2
...
```

but i guess that can get weirdly lost somewhere. This works, i just wanted to know if we have some weird underlying issue somewhere but incompatible versions are understandable.